src/main/java/mate/academy/service/impl/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,41 @@
+package mate.academy.service.impl;
+
+import java.util.Optional;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.exception.RegistrationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.AuthenticationService;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        Optional<User> userFromDbOptional = userService.findByEmail(email);
+        User user = userFromDbOptional.get();
+        if (userFromDbOptional.isEmpty()
+                || !HashUtil.hashPassword(password, user.getSalt()).equals(user.getPassword())) {
+            throw new AuthenticationException("Can`t authenticate user. We don`t have this user.");
```

Comment:
You don't have, or service? Write a message from the service retrospective 
```suggestion
            throw new AuthenticationException("Can`t authenticate user. Check email or password.");
```